controllers/host/host_controller.go

```diff
@@ -979,12 +979,41 @@ func (r *HostReconciler) CompareEnabledAttributes(in *starlingxv1.HostProfileSpe
 
 	if utils.IsReconcilerEnabled(utils.FileSystemSizes) {
 		if in.Storage != nil && in.Storage.FileSystems != nil {
-			if other.Storage == nil {
+			if other.Storage == nil || other.Storage.FileSystems == nil {
 				return false
 			}
 
-			if !in.Storage.FileSystems.DeepEqual(other.Storage.FileSystems) {
-				return false
+			// Special case: 'ceph' filesystem must be ignored for All-in-one Duplex.
+			if system_info.SystemType == cloudManager.SystemTypeAllInOne && system_info.SystemMode != cloudManager.SystemModeSimplex {
+				for _, fsInfo := range *other.Storage.FileSystems {
+					// Skip ceph filesystem
+					if fsInfo.Name == "ceph" {
+						continue
+					}
+
+					// Compare 'in' and 'other' fileystems
+					found := false
+					for _, fs := range *in.Storage.FileSystems {
+						if fs.Name != fsInfo.Name {
+							continue
```

Comment:
Can you extract the inner `for` loop to a method: findFileSystemByName(*in.Storage.FileSystems, fsInfo.Name) and we can do something like:

if matchedFileSystem == nil || matchedFs.Size != fsInfo.Size {
        return false
}

this will eliminate the need for the flag and make the code more modular. We want to avoid expanding the method’s scope, as that would make it more complex to understand.

Reply:
I can do that.
I did this way copying from another part of the code. My intention was to maintain the same schema.

Reply:
@fsanches-wr This was a suggestion to improve the code. Since it’s not a major change to the method, I’ll go ahead and merge it.